eng/scripts/automation/Invoke-CadlDataPlaneGenerateSDKPackage.ps1

```diff
@@ -45,24 +39,57 @@ Invoke-CadlDataPlaneGenerateSDKPackage.ps1 -service anomalydetector -namespace A
 
 #>
 param (
-  [string]$service,
-  [string]$namespace,
-  [string]$sdkPath,
-  [string]$cadlRelativeFolder,
+  [string]$sdkFolder,
+  [string]$cadlSpecDirectory,
   [string]$specRoot = "",
-  [string]$repo="",
-  [string]$commit="",
-  [string]$additionalSubDirectories="" #additional directories needed, separated by semicolon if more than one
+  [string]$repo = "Azure/azure-rest-api-specs",
+  [string]$commit = "",
+  [string]$additionalSubDirectories="", #additional directories needed, separated by semicolon if more than one
+  [switch]$help
+)
+if ($help) {
+  Write-Host("Usage:
+Invoke-CadlDataPlaneGenerateSDKPackage.ps1 -sdkFolder <sdk-folder-path> -cadlSpecDirectory <relativeCadlProjectFolderPath> [-commit <commitId>] [-repo <specRepo>] [-specRoot <specRepoRootPath>] [-additionalSubDirectories <relativeFolders>]
+
+Options:
+-sdkFolder [Required] take the address of the sdk folder in azure-sdk-for-net repo. e.g. /home/azure-sdk-for-net/sdk/anomalyDetector/Azure.AI.AnomalyDetector
+-cadlSpecDirectory [Required] takes the relative path of the cadl project folder in spec repo. e.g. specification/cognitiveservices/AnomalyDetector
+-additionalSubDirectories [Optional] takes the relative paths of the additional directories needed by the cadl project
+-commit takes the git commit hash  (e.g. ac8e06a2ed0fc1c54663c98f12c8a073f8026b90)
+-repo [Optional] takes the `<owner>/<repo>` of the REST API specification repository. (e.g. Azure/azure-rest-api-specs). The default is Azure/azure-rest-api-specs
+-specRoot takes the file system path of the spec repo. e.g. /home/azure-rest-api-specs
+
+hint: You need to provide the cadl project path either (`-commit`, `-repo`) pair to refer to an URL path of the cadl project or `-specRoot` to refer to local file system path. If you provide both, `-specRoot` will be ignored.
+"
 )
+  exit 0;
+}
+
+# validate input parameter
+if (!$sdkFolder -or !$cadlSpecDirectory) {
+  Throw "One of required parameters (sdkFolder, cadlSpecDirectory) is missing. Please use -help to see the usage."
+}
+
+if ((!$commit) -And !$specRoot) {
+  Throw "The cadl project path is not provided. You need to provide either (-commit, -repo) pair to refer to URL path or -specRoot to refer to local file system path."
+}
 . (Join-Path $PSScriptRoot GenerateAndBuildLib.ps1)
-$sdkPath = Resolve-Path $sdkPath
+
+# $sdkFolder = Resolve-Path $sdkFolder
+$sdkFolder = $sdkFolder -replace "\\", "/"
+$sdkFolder = $sdkFolder.TrimEnd('/')
 # Generate dataplane library
 $outputJsonFile = "newpackageoutput.json"
+if ( $sdkFolder -match "(?<sdkPath>.*)/sdk/(?<serviceDirectory>.*)/(?<namespace>.*)" ) {
+  $sdkPath = Resolve-Path $matches["sdkPath"]
+  $service = $matches["serviceDirectory"]
+  $namespace = $matches["namespace"]
+}
 New-CADLPackageFolder `
   -service $service `
```

Comment:
Below did we want to call Cadl-Project-Generate.ps1 instead of the GenerateCode directly? @m-nash where would you expect use to call the common generate script in this workflow?

Reply:
I think we can write an common interface GenerateSDK which will automate everything. User does not need to manually create/update cadl-location.yaml also.

Example
```
function GenerateSDK([string]$projectDirectory, [string]$cadlSpecDirectory, [string]$commit, [string]$repo, [string]$localSpecRootDirectory) {
  CreateorUpdateCadlLocationFile
  cadl-project-sync
  cadl-Project-Generate
}
```

Reply:
We wouldn't expect customers to ever use those scripts directly they should simply call `dotnet build /t:GenerateCode`.  For other languages like python they don't have a common build tool to integrate with so customers will directly call the sync / generate scripts for now.